src/Evolution/Systems/Cce/AnalyticSolutions/SphericalMetricData.hpp

```diff
@@ -47,6 +47,22 @@ struct SphericalMetricData : public WorldtubeData {
   explicit SphericalMetricData(const double extraction_radius) noexcept
       : WorldtubeData{extraction_radius} {}
 
+  /*!
+   * Computes and returns by pointer the Jacobian
+   * \f$\partial x_{\mathrm{Cartesian}}^j /
+   * \partial x_{\mathrm{spherical}}^i\f$
+   */
+  void jacobian(gsl::not_null<SphericaliCartesianJ*> jacobian,
+                size_t l_max) const noexcept;
+
+  /*!
+   * Computes and returns by pointer the first radial derivative of the
+   * Jacobian: \f$\partial_r (\partial x_{\mathrm{Cartesian}}^j /
+   * \partial x_{\mathrm{Spherical}}^i)\f$
+   */
+  static void dr_jacobian(gsl::not_null<SphericaliCartesianJ*> dr_jacobian,
+                          size_t l_max) noexcept;
+
```

Comment:
(picking a line at random) I think none of the functions in `SphericalMetricData` are currently tested, is that correct? At least I couldn't find tests in `tests/Evolution/Systems/Cce/AnalyticSolutions`. Is the idea that these functions are tested with derived classes of `SphericalMetricData`? Perhaps you could add a dummy derived class in a test to add Python random-value tests for these implementations?

Reply:
They are not currently tested independently, but all of the features are tested either directly or indirectly by their derived classes -- these new features will be part of the test for `TeukolskyWave`.

Unfortunately, the python tests won't work with these data generation tools without significant alteration or re-development, as these generate the collocation point data from the `l_max` (and nothing else), so would require putting in python routines for obtaining those points from libsharp, and for the python tests to not act on elements of vectors separately.

Reply:
I see, thanks for explaining. I'm not going to block on this issue if @nilsdeppe and/or @kidder who have reviewed more of the CCE code and are therefore more familiar with the context of this code are OK with merging this PR without tests. Here are my concerns though:
- I don't trust myself checking all indices and terms in these equations, which is why a suite of sanity-checks and a comparison to an independent (and often more readable) Python implementation greatly increases my confidence that the implementation is correct. If I understand correctly the first commit fixes errors in the implementation that ideally would have been caught in a test when this code was merged initially.
- If code is not easily unit-testable it's often an indication that it should be implemented more granularly. Here I think the `lmax` you're feeding in to these functions gets translated to collocation points and is used to compute angular SWSH derivatives, with all other code in these functions building on top of that. So one possibility would be to split the code into functions that call into libsharp (that you either don't have to test because they are tested elsewhere, or that at least don't require a Python comparison), and the new code that computes all these terms and that's fairly important to test. Then all the new code in these functions you can compare to a Python implementation by assuming a few hard-coded collocation points, for example.

Reply:
The code and design is fine given the way CCE in general works. It's not nearly as pointwise as evolution or elliptic problems.